andela/ah-django-unchained
# #162948842 Have CRUD on Articles Resource

#### What does this PR do?
Have CRUD on Articles resource
#### Description of Task to be completed?

`POST api/articles/`

`GET api/articles/`

`GET api/articles/:slug`

`PUT api/articles/:slug`

`PUT api/articles/delete/:slug`

#### How should this be manually tested?

- Clone the repo locally
- cd ah-django-unchained
- git checkkout ft-crud-on-articles-162948842
- get started with the app as from the README section
- to add an article ensure you obtain a token from logging in or signing up if not signed up
- Using the `POST api/articles/` endpoint, have the following JSON object. Note that image uploads are optional.
```
{
    "title": "My story",
    "description": "I am very happy",
    "body": "A story about me"
}
```
<img width="1136" alt="post article" src="https://user-images.githubusercontent.com/31338414/51385132-66958a80-1b2f-11e9-8f1e-31cbd66e4850.png">

- Using the `GET api/articles/`, fetch all articles. Only those that have `is_displayed=True` will be displayed as we assume the others have been deleted.
<img width="1034" alt="get all articles" src="https://user-images.githubusercontent.com/31338414/51385308-e91e4a00-1b2f-11e9-8bf6-303480c1462f.png">

- Using the `GET api/articles/:slug` endpoint we are able to fetch a single article depending on the name of the slug
<img width="1031" alt="get single article" src="https://user-images.githubusercontent.com/31338414/51385398-37cbe400-1b30-11e9-8108-f766dbf4c6ce.png">

- Using the `PUT api/articles/:slug` we can edit an article. Use the JSON body below
```
{
    "title": "My story",
    "description": "I am very happy",
    "body": "A story about me"
}
```

- Using the `PUT api/articles/delete/:slug`, we can delete an article by setting its `is_displayed` field to `False`
```
{
    "is_displayed": False
}
```

#### What are the relevant pivotal tracker stories?
[#162948842](https://www.pivotaltracker.com/story/show/162948842)
